gcp/apis/compute/v1alpha1/network_types_test.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestGCPNetworkSpec_IsSameWith(t *testing.T) {
```

Comment:
I agree that mixed caps makes sense. But I am not sure whether this convention goes for naming of test functions, too. Goland automatically puts underscore for the functions that tests a function of a struct, i.e. `TestStruct_FuncName`. I see that consul also does that, as seen here towards the end https://ieftimov.com/post/testing-in-go-naming-conventions/

To me, it feels more intuitive to have `TestGCPNetworkSpec_IsSameWith`, `TestGCPNetworkSpec_AnotherFunction` etc. than `TestGCPNetworkSpecIsSameWith` and `TestGCPNetworkSpecAnotherFunction`. It implies that the function and owner struct better and it could go really long when you actually concatenate two names of individual code pieces (struct + function) without any separator.

Reply:
I'd still prefer to omit the underscore; I don't see why test files would be exempt from this convention. That said, I don't want to block the PR on the naming of tests.